r/src/arrow_types.h

```diff
@@ -37,25 +37,74 @@ struct NoDelete {
   inline void operator()(T* ptr){};
 };
 
+namespace arrow {
+namespace r {
+struct symbols {
+  static SEXP units;
+  static SEXP xp;
+};
+}  // namespace r
+}  // namespace arrow
+
 namespace Rcpp {
+namespace internal {
+
+template <typename Pointer>
```

Comment:
There's some black magic in this file. Might be worth a blog post explaining how to deal with different kinds of smart pointers using external pointers in R6 classes

Reply:
Yes definitely on my list of things. This is close to Rcpp's metal so that might be interesting to that community as well.